src/main/java/org/wildfly/security/auth/provider/ldap/LdapSecurityRealm.java

```diff
@@ -115,7 +115,46 @@ public ModifiableRealmIdentity getRealmIdentityForUpdate(String name, final Prin
 
     @Override
     public Iterator<ModifiableRealmIdentity> getRealmIdentityIterator() throws RealmUnavailableException {
-        return null;
+        if (identityMapping.iteratorFilter == null) {
+            throw log.ldapRealmNotConfiguredToSupportIteratingOverIdentities();
+        }
+
+        log.debugf("Trying to iterate over identities.");
+        DirContext context = null;
+        NamingEnumeration<SearchResult> result;
+
+        try {
+            context = dirContextFactory.obtainDirContext(null);
+
+            result = context.search(identityMapping.searchDn, identityMapping.iteratorFilter,
+                    identityMapping.iteratorFilterArgs, createSearchControls(identityMapping.rdnIdentifier));
+
+        } catch (NamingException e) {
+            throw log.ldapRealmIdentitySearchFailed(e);
+        } finally {
+            dirContextFactory.returnContext(context);
```

Comment:
Is this definitely Ok that the context is returned but we are still working with the results?  Are the results eagerly loaded?